### PR TITLE
CMake files: some minor cleaning up (#83)

### PR DESCRIPTION
**Pull request by agarny**

### CODE REVIEW

**Comment by nickerso** [comment]
Its important to include the actual issue being addressed in the description of the pull request (which should probably never be empty) so that the GitHub dependency thingy usefully links between issues and pull requests. 

This pull request addresses #83.

**Comment by nickerso** [comment]
This all looks good to me.

File: CMakeLists.txt [hunk 1]
```diff
@@ -40,55 +40,55 @@ endif()
...
+if(UNIT_TESTS)
+  set(LIBCELLML_UNIT_TESTS "${UNIT_TESTS}" CACHE BOOL ${_PARAM_ANNOTATION} FORCE)
 endif()
 
 if(LIBCELLML_COVERAGE OR LIBCELLML_MEMCHECK)
   set(LIBCELLML_UNIT_TESTS ON CACHE BOOL ${_PARAM_ANNOTATION} FORCE)
 endif()
 
-# TWAE ==> LIBCELLML_WARNINGS_AS_ERRORS -- Note: This excludes third party code, where warnings are never treated as errors.
+# TREAT_WARNINGS_AS_ERRORS ==> LIBCELLML_WARNINGS_AS_ERRORS -- Note: This excludes third party code, where warnings are never treated as errors.
```
Comment (hsorby) [inline CMakeLists.txt]: This was shortened on purpose because it is only accessible through the command line.  I  prefer usability over looking nice.

**Comment by agarny** [comment]
> Its important to include the actual issue being addressed in the description of the pull request (which should probably never be empty) so that the GitHub dependency thingy usefully links between issues and pull requests.

Sorry, I mentioned it in the subject of the PR, but I appreciate that a short description would have been useful, if anything so that we would have a proper link to the referenced issue.

File: CMakeLists.txt (hunk 1 above)
Comment (agarny) [inline CMakeLists.txt]: It's not about looking nice, but about being consistent. Anyway, I have shortened it a bit. Hopefully, you find it more usable that way (not that this option should be used that often anyway).

**Comment by agarny** [comment]
@nickerso, @hsorby, OK to merge?

**Comment by nickerso** [comment]
ok with me, hoping @hsorby gets a chance to review before merging it in.

**Comment by hsorby** [comment]
Sorry, I have been twisting and turning on the command line WARNINGS_AS _ERRORS.  finally I have decided on my opinion and it is ...

I prefer the TWAE shortcut for the command line version and the documentation has the explanation, which you would have to read if you wanted to make use of it.  As the default is on you would have to actively turn it off and to know the option is there you would have to do some reading.  So I would like to see the original version and not the long version given in this PR.

**Comment by agarny** [comment]
It's not a majority vote and I don't buy the argument, but nevermind. Going with @hsorby so we can finally merge this PR.

**Comment by jonc125** [comment]
I can see benefits for having a command line shortcut - in Chaste we have both long & shortcut forms available. But as @agarny says it's a minor issue; let's merge!

File: CMakeLists.txt [hunk 2]
```diff
@@ -60,11 +60,11 @@ if(LIBCELLML_COVERAGE OR LIBCELLML_MEMCHECK)
   set(LIBCELLML_UNIT_TESTS ON CACHE BOOL ${_PARAM_ANNOTATION} FORCE)
 endif()
 
-# WARNINGS_AS_ERRORS ==> LIBCELLML_WARNINGS_AS_ERRORS -- Note: This excludes third party code, where warnings are never treated as errors.
+# TWAE ==> LIBCELLML_WARNINGS_AS_ERRORS -- Note: This excludes third party code, where warnings are never treated as errors.
```
Comment (jonc125) [inline CMakeLists.txt]: Should be `LIBCELLML_TREAT_WARNINGS_AS_ERRORS` in this comment... alas for the nitpicking!

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -105,7 +105,7 @@ set_target_properties(cellml PROPERTIES
   CXX_VISIBILITY_PRESET hidden
   VISIBILITY_INLINES_HIDDEN 1)
 
-if(LIBCELLML_TREAT_WARNINGS_AS_ERRORS)
+if(LIBCELLML_WARNINGS_AS_ERRORS)
```
Comment (nickerso) [inline src/CMakeLists.txt]: needs reverting?

File: tests/CMakeLists.txt [hunk 4]
```diff
@@ -37,7 +37,7 @@ foreach(TEST ${LIBCELLML_TESTS})
     CXX_STANDARD_REQUIRED ON
     FOLDER tests)
 
-#  if(LIBCELLML_TREAT_WARNINGS_AS_ERRORS)
+#  if(LIBCELLML_WARNINGS_AS_ERRORS)
```
Comment (nickerso) [inline tests/CMakeLists.txt]: needs reverting?

File: CMakeLists.txt (hunk 2 above)
Comment (agarny) [inline CMakeLists.txt]: My bad, I reverted things not realising that there was originally a 'mistake'.

File: src/CMakeLists.txt (hunk 3 above)
Comment (agarny) [inline src/CMakeLists.txt]: Makes me wonder whether we shouldn't have a way to test that... Oh well...

**Pull request merged**